shm/cargo.cpp

```diff
@@ -0,0 +1,33 @@
+#include "cargo.hpp"
+#include <iostream>
+
+
+Cargo::Cargo(const std::string& name, size_t amount, size_t basePrice, std::shared_ptr<Time> time)
+    : name_(name)
+    , amount_(amount)
+    , basePrice_(basePrice)
+    , time_(time)
+    {
+        if (time_) {
+            time_->attachObserver(this);
```

Comment:
You never detach observer. This will cause crash when this object will be deleted